clusterloader2/pkg/imagepreload/imagepreload.go

```diff
@@ -165,6 +164,7 @@ func (c *controller) hasPreloadedImages(node *v1.Node) bool {
 	for _, nodeImg := range node.Status.Images {
 		nodeImages = append(nodeImages, nodeImg.Names...)
 	}
+	klog.V(3).Infof("Found the following images on node: %s", nodeImages)
```

Comment:
Hmmm, I don't see any logs for this line. That means we didn't get any node update events. We can exclude the situation where there were no node updates as the node-problem-detectors updates the nodes periodically. The only explanation is we set up the watch/informer incorrectly. Let me dig into this. 

In general, I'm not sure if this code was ever tested, so the bug can be anywhere.

Reply:
Oh, the logs ARE there - https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/perf-tests/1654/pull-perf-tests-clusterloader2/1350026427449217024/

I was looking at the old run. That means the ctr pull image in the container is not updating the images on the host. Let me play a bit with that.